app/modules/conversations/conversations_router.py

```diff
@@ -160,11 +160,21 @@ async def share_chat(
 ):
     service = ShareChatService(db)
     try:
-        shared_conversation = await service.share_chat(
-            request.conversation_id, request.recipientEmails
-        )
+        shared_conversation = await service.share_chat(request.conversation_id, request.recipientEmails, request.visibility)
         return ShareChatResponse(
             message="Chat shared successfully!", sharedID=shared_conversation
         )
     except ShareChatServiceError as e:
         raise HTTPException(status_code=400, detail=str(e))
+
+@router.get("/conversations/{conversation_id}/shared-emails", response_model=List[str])
+async def get_shared_emails(
+    conversation_id: str,
+    db: Session = Depends(get_db),
+):
```

Comment:
Where is the authorisation? 
Add auth and make sure that the get emails endpoint returns 401 when a user who is not the conversation owner tries to request them